tests/libbpfgo.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+#
+# This test is executed by github workflows inside the action runners
+#
+
+info() {
+    echo -n "INFO: "
+    echo "$@"
+}
+
+error_exit() {
+    echo -n "ERROR: "
+    echo "$@"
+    exit 1
+}
+
+TRACEE_DIR="$(realpath ./)"
+TEST_DIR="$(dirname "$(realpath "${0}")")"
+LIBBPFGO_DIR="${TEST_DIR}/libbpfgo"
+LIBBPFGO_REPO="https://github.com/aquasecurity/libbpfgo.git"
+
+if [[ -d "${LIBBPFGO_DIR}" ]]
+then
+    info "Reusing ${LIBBPFGO_DIR}"
+else
+    git clone "${LIBBPFGO_REPO}" "${LIBBPFGO_DIR}" || error_exit "could not clone ${LIBBPFGO_REPO}"
+fi
+
+echo "replace github.com/aquasecurity/libbpfgo => ${LIBBPFGO_DIR}" >> "${TRACEE_DIR}"/go.mod
```

Comment:
Triggering go mod vendor would fetch all dependencies. I assume we can just rely on go getting `github.com/aquasecurity/libbpfgo@main` instead of cloning it. Let me test it.

Reply:
We can use this [approach](https://github.com/geyslan/tracee/blob/d6dae699df6a889bd7d99f34b6b016169da5ba11/.github/workflows/test-daily.yaml#L28-L34), directly in the workflow (detached from scripts):

```yaml
      - name: Compile Tracee
        run: |
          make ./dist/libbpf/libbpf.a
          CGO_CFLAGS="-I $(realpath "./dist/libbpf")" \
            CGO_LDFLAGS="-lelf -lz $(realpath "./dist/libbpf/libbpf.a")" \
            go get github.com/aquasecurity/libbpfgo@main
          STATIC=1 make
```

Or continue putting that logic inside the proposed `tests/libbpfgo.sh` script.

Even better, creating a `test-upstream-libbpfgo` rule to Makefile. WDYT?

Action tested with yaml changes above: https://github.com/geyslan/tracee/runs/7845940102?check_suite_focus=true

Reply:
I like that approach (`make test-upstream-libbpfgo`) much better, let's go with that!

Reply:
@grantseltzer We had to keep the external script due to unordered Makefile rule commands. But the effort of put that rule was worth it, since we can reuse the env vars from Makefile and trigger the test locally too.

https://github.com/geyslan/tracee/runs/7862082846?check_suite_focus=true&ts=8#step:4:2

https://github.com/geyslan/tracee/runs/7862082846?check_suite_focus=true&ts=8#step:4:44